src/BullOak.Repositories/Maybe.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace BullOak.Repositories
+{
+    public struct Maybe<T>
```

Comment:
Have you considered Optional package?
<https://github.com/nlkl/Optional>,  <https://www.nuget.org/packages/Optional/>

Reply:
For this project it is extremely important to limit the number of dependencies and because of the very limited requirements I did not try to find any packages that does this.

Nice project though.